### ISSUE TITLE
Track down sites suggesting to blindly preload

### ISSUE THREAD

**Issue by lgarron** [chromium/hstspreload.appspot.com#68]
Latest one: 
- https://docs.nextcloud.com/server/9/admin_manual/configuration_server/harden_server.html#enable-http-strict-transport-security

**Comment by lgarron** [comment]
https://github.com/roots/trellis defaults to `preload`.

**Comment by lgarron** [comment]
PR set to `twitter/secureheaders` at https://github.com/twitter/secureheaders/pull/310

**Comment by lgarron** [comment]
The hstspreload.org site now also has a section about this: https://hstspreload.org/#opt-in

**Comment by lgarron** [comment]
https://www.section.io/blog/using-varnish-to-tweak-your-http-responses/
<img width="1105" alt="screen shot 2017-01-18 at 18 36 32" src="https://cloud.githubusercontent.com/assets/248078/22091700/0dbbcf76-ddad-11e6-8d05-cca86e8a80ce.png">

The [relevant gist](https://gist.github.com/section-io-gists/39e0ddc31cdd7ca8ae33#file-security_headers-vcl) is owned by @section-io-gists

**Comment by glennslaven** [comment]
Hi @lgarron,

Thanks for pointing this out, we've updated the gist to remove the preload directive.

**Comment by lgarron** [comment]
> Thanks for pointing this out, we've updated the gist to remove the preload directive.

Looks good, thanks!
(You may also want to remove the trailing semicolon, but that doesn't affect correctness.)

**Comment by graingert** [comment]
@lgarron maybe the header should be changed to:
```
preload, SHA3(`${domain} I have read and understood https://hstspreload.org/#information`)
```

**Comment by lgarron** [comment]
> @lgarron maybe the header should be changed to:

Then someone will find a way to automate it and shoot their users in the foot. :-P

More seriously, a site-specific confirmation is not a bad idea, but
1) it requires extra configuration for infrastructure shared across multiple hosts (a specific value per host, instead of ≈ a boolean)
2) It requires changing the header convention. This is non-trivial, and I don't want to change or add anything other than `preload` without a spec. It also adds an additional special case for old domains.

It would probably be a good idea to discuss this at a meetup this year, which I would like to organize again once we've automated scanning and pruning.

**Comment by lgarron** [comment]
@graingert: Actually, would you mind filing a separate issue for that idea, so we can keep track of any progress on it in one place?

**Comment by graingert** [comment]
@lgarron 1. no because you can provide a set of SHA3(domain + edu-nonce) and as long as your domain is in the set you win

win = preload

**Comment by graingert** [comment]
Those that know enough to automate it, know not to

**Comment by lgarron** [comment]
> @lgarron 1. no because you can provide a set of SHA3(domain + edu-nonce) and as long as your domain is in the set you win

Bloating headers is not a good idea. :-/

**Comment by graingert** [comment]
HPACK means it's fine

**Comment by lgarron** [comment]
> HPACK means it's fine

I don't think this is an appropriate assumption to make on behalf of all sites and clients.

In any case, I think a single hash for just the current domain is reasonable to ask for, and rolling it out won't be any harder from our end.

**Comment by lgarron** [comment]
Talks about preloading but doesn't mention how to submit to the list: https://blog.stackpath.com/glossary/hsts/

**Comment by lgarron** [comment]
https://varvy.com/pagespeed/hsts.html

**Comment by Malvoz** [comment]
I think https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security could need some clarification around `preload` and the submission list.
